src/core/border_router/routing_manager.cpp

```diff
@@ -1973,47 +1961,27 @@ uint32_t RoutingManager::DiscoveredPrefixTable::Entry::CalculateExpireDelay(uint
 //---------------------------------------------------------------------------------------------------------------------
 // OmrPrefix
 
-void RoutingManager::OmrPrefix::Init(const Ip6::Prefix &aPrefix, RoutePreference aPreference)
-{
-    mPrefix     = aPrefix;
-    mPreference = aPreference;
-}
-
-void RoutingManager::OmrPrefix::InitFrom(NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig)
+void RoutingManager::OmrPrefix::SetFrom(const NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig)
 {
-    Init(aOnMeshPrefixConfig.GetPrefix(), aOnMeshPrefixConfig.GetPreference());
+    mPrefix     = aOnMeshPrefixConfig.GetPrefix();
+    mPreference = aOnMeshPrefixConfig.GetPreference();
 }
 
-bool RoutingManager::OmrPrefix::IsFavoredOver(const OmrPrefix &aOther) const
+bool RoutingManager::OmrPrefix::IsFavoredOver(const NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig) const
 {
     // This method determines whether this OMR prefix is favored
-    // over `aOther` prefix. A prefix with higher preference is
+    // over another prefix. A prefix with higher preference is
     // favored. If the preference is the same, then the smaller
     // prefix (in the sense defined by `Ip6::Prefix`) is favored.
 
-    return (mPreference > aOther.mPreference) || ((mPreference == aOther.mPreference) && (mPrefix < aOther.mPrefix));
-}
-
-RoutingManager::OmrPrefix::InfoString RoutingManager::OmrPrefix::ToString(void) const
-{
-    InfoString string;
-
-    string.Append("%s (prf:", mPrefix.ToString().AsCString());
+    bool isFavored = (mPreference > aOnMeshPrefixConfig.GetPreference());
```

Comment:
nit: add assertion that `aOnMeshPrefixConfig` is a valid OMR prefix.

I was thinking that we need to check `aOnMeshPrefixConfig` is a valid OMR prefix before comparing it, but looks like we already checked it outside this method in `EvaluateOmrPrefix()`. Another option is that we merge `if (!IsValidOmrPrefix(onMeshPrefixConfig) || !onMeshPrefixConfig.mPreferred)` into this method so that it accepts all on-mesh prefixes.

Reply:
Adding `OT_ASSERT()` sounds good.